dln.c

```diff
@@ -25,6 +25,21 @@ static void dln_loaderror(const char *format, ...);
 #include "internal.h"
 #include "internal/compilers.h"
 
+/* Only check ABI version for development Ruby. Released versions of Ruby
+ * should guarantee ABI compatibility between patch versions. */
+/* Windows does not support weak symbols so rb_abi_version will not exist
+ * in the shared library. */
+#if defined(RUBY_DEVEL) && !defined(_WIN32) && !defined(__MINGW32__)
+# define RUBY_DLN_CHECK_ABI
+
+/* Include header files as dependency for dln.c */
+# include "ruby/ruby.h"
+# include "ruby/atomic.h"
+# include "ruby/debug.h"
+# include "ruby/encoding.h"
+# include "ruby/fiber/scheduler.h"
```

Comment:
Seems unnecessary.
At least, "ruby/ruby.h" is included already.

Reply:
It's to include as a dependency in `common.mk` so `dln.c` gets rebuilt anytime any of these headers change.

Reply:
I think version.c is the better file for such volatile info.

Reply:
Even if `dln.c` depends on the all headers, `abi.h` doesn't seem to be updated after it is created once.

Reply:
I'm not sure how to repro this. I did the following:

1. I ran `make miniruby`. I get the output `include/ruby/internal/abi.h updated`.
2. I changed a comment in a header file (`include/ruby/ruby.h`).
3. I ran `make miniruby` again.  I get the output `include/ruby/internal/abi.h updated`.